### ISSUE TITLE
How to -- run a function after scrolling

### ISSUE THREAD

**Issue by bobabarbanel** [olifolkerd/tabulator#797]
I need to add a class to some cells in some rows which I can compute at any point. The computation depends on comparing values in one row against the values in the previous row. This is essentially the grouping function. I cannot use grouping at this time because addRow fails there at this time. So I am attempting to indicate the end of a group with a simple border on the cells of the previous row.

The Tabulator does not instantiate cells in rows until that row is drawn - so - I cannot add the class unless I can detect that a row has been made visible. I have a row formatter function but it is not called for every row at all times.

I think I need a "flow-diagram" with events so I can tell what gets called when? The main issue at this point is that when the table is scrolled and cells get allocated, I need to determine if those cells need my border marks.

**Comment by olifolkerd** [comment]
Hey @bobabarbanel 

Thanks for getting in touch,

In this case you want to be using formatters to do this. if you are applying this accross multiple cells in a row then i would suggest using the ***rowFormatter***

have a look at the [Row Formatter Documentation](http://tabulator.info/docs/3.3#ajax) for more information.

```js
$("#example-table").tabulator({
    rowFormatter:function(row){
        //row - row component

        var cells= row.getCells(); //get all cells in row

        //perform an action on cells      
    },
});
```

**Comment by bobabarbanel** [comment]
Yes, but it does not get called on every row with All changes. 

Bob Abarbanel
+1 206-499-1443
(Pleas forgive 21st speling errors. 😜)

**Comment by olifolkerd** [comment]
what do you mean by all changes?

**Comment by bobabarbanel** [comment]
When I add a row, for example, the formatted is NOT called on every row in the table. I need to process every row to detect value changes across the whole table. 

Bob Abarbanel
+1 206-499-1443
(Pleas forgive 21st speling errors. 😜)

**Comment by bobabarbanel** [comment]
One can call the rowformatter explicitly. That is the solution I am using now. 

Bob Abarbanel
+1 206-499-1443
(Pleas forgive 21st speling errors. 😜)

**Comment by bobabarbanel** [comment]
To be able to impact the cell’s element’s styles I have to ignore the virtual dom. Having done that, I can change the styles as the data changes. But I am having to call the row formatter on my own.

Bob Abarbanel
+1 206.499.1443

**Comment by olifolkerd** [comment]
Hey bob,

i see what you mean, 

In that case i would suggest using the ***rowAdded*** callback as a trigger point for this you could then call the ***getRows*** function to get an array of all the rows in the table and iterate through them.

as an enhancement, in the next release, i will add a ***reformat*** function to the row component to allow users to trigger the row formatter on that row.

Cheers

Oli

**Comment by olifolkerd** [comment]
Hey @bobabarbanel 

You will be happy i have just pushed this feature to the 3.4 branch, you will be able to trigger a reformat of the row by calling the ***reformat*** function on the row component:
```js
row.reformat();
```

This will be available in the 3.4 release in a couple of weeks.

Cheers

Oli :)

**Comment by olifolkerd** [comment]
Tabulator version 3.4 has now been officially release!

You can find details of the new release in the [News Section](http://tabulator.info/news)

Full documentation of all the new features can be found [Here](http://tabulator.info/docs/3.4)

Cheers

Oli :)
